edgelet/provisioning/src/provisioning.rs

```diff
@@ -20,15 +20,47 @@ use edgelet_http::client::{Client as HttpClient, ClientImpl};
 use edgelet_utils::log_failure;
 use hsm::TpmKey as HsmTpmKey;
 use log::Level;
+use sha2::{Digest, Sha256};
+use std::str::FromStr;
 
 use crate::error::{Error, ErrorKind};
 
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub enum ReprovisioningStatus {
+    DeviceDataNotUpdated,
+    DeviceDataUpdated,
+    InitialAssignment,
+    DeviceDataMigrated,
+    DeviceDataReset,
+}
+
+impl FromStr for ReprovisioningStatus {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<ReprovisioningStatus, ()> {
+        // TODO: check with DPS substatus value for DeviceDataUpdated when it is implemented on service side
+        match s {
+            "deviceDataMigrated" => Ok(ReprovisioningStatus::DeviceDataMigrated),
+            "deviceDataReset" => Ok(ReprovisioningStatus::DeviceDataReset),
+            _ => Ok(ReprovisioningStatus::InitialAssignment),
```

Comment:
Also add a case specifically for `"initialAssignment" => Ok(Reprovisioning::InitialAssignment)` so that it clearly separates the actual value from the fallback case for anyone reading the code.